pkg/types/config/cluster.go

```diff
@@ -28,94 +25,29 @@ const (
 // Platform indicates the target platform of the cluster.
 type Platform string
 
-// UnmarshalYAML unmarshals and verifies the platform.
-func (p *Platform) UnmarshalYAML(unmarshal func(interface{}) error) error {
-	var data string
-	if err := unmarshal(&data); err != nil {
-		return err
-	}
-
-	platform := Platform(data)
-	switch platform {
-	case PlatformAWS, PlatformLibvirt, PlatformOpenStack:
-	default:
-		return fmt.Errorf("invalid platform specified (%s); must be one of %s", platform, []Platform{PlatformAWS, PlatformLibvirt, PlatformOpenStack})
-	}
-
-	*p = platform
-	return nil
-}
-
-var defaultCluster = Cluster{
-	AWS: aws.AWS{
-		Endpoints:    aws.EndpointsAll,
-		Profile:      aws.DefaultProfile,
-		Region:       aws.DefaultRegion,
-		VPCCIDRBlock: aws.DefaultVPCCIDRBlock,
-	},
-	Libvirt: libvirt.Libvirt{
-		Network: libvirt.Network{
-			IfName: libvirt.DefaultIfName,
-		},
-	},
-	Networking: Networking{
-		MTU:         "1480",
-		PodCIDR:     "10.2.0.0/16",
-		ServiceCIDR: "10.3.0.0/16",
-		Type:        tectonicnetwork.NetworkFlannel,
-	},
-	OpenStack: openstack.OpenStack{
-		Region:           openstack.DefaultRegion,
-		NetworkCIDRBlock: openstack.DefaultNetworkCIDRBlock,
-	},
-}
-
 // Cluster defines the config for a cluster.
 type Cluster struct {
```

Comment:
Can we just rename this struct to `TFVars` as that's what this is. can be separate PR; but we are a lot about this struct so I thought name might also be in the area :P.

Reply:
> Can we just rename this struct to `TFVars`...

I've pushed 5882b2e -> 4d47856 to move this out of `pkg/types` too and simplify this package's [current API][1] to a single function:

```console
$ godoc ./pkg/tfvars/
PACKAGE DOCUMENTATION

package tfvars
    import "./pkg/tfvars/"

    Package tfvars converts an InstallConfig to Terraform variables.

FUNCTIONS

func TFVars(cfg *types.InstallConfig, bootstrapIgn string, masterIgns []string, workerIgn string) ([]byte, error)
    TFVars converts the InstallConfig and Ignition content to
    terraform.tfvar JSON.

SUBDIRECTORIES

	aws
	libvirt
	openstack

```

[1]: https://godoc.org/github.com/openshift/installer/pkg/types/config